src/Graph.ts

```diff
@@ -454,6 +456,15 @@ export default class Graph {
 			module.dependencies.forEach(depModule => {
 				if (!depModule.isExternal) {
 					visit(<Module>depModule, seen);
+					const depId = relativeId(depModule.id);
+					cycles.forEach(cycle => {
+						if (cycle.path[depId] === null) {
```

Comment:
The complexity of this loop does worry me.... cyclical code shouldn't get a performance penalty. Ideally as much of the looping could be moved into the warning function itself, with only markers provided from this main iteration loop (eg parent information etc).

Having these extra loops within the main visit function doesn't seem ideal.

Reply:
This is only looping over the list of cycles, which usually should be 0 or 1.